target/linux/ramips/dts/mt7620n_hootoo_tm05.dts

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7620n.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "hootoo,tm05", "ralink,mt7620n-soc";
+	model = "HooToo TM05";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &ethernet;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power {
+			label = "tm05:blue:power";
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		wifi {
+			label = "tm05:green:wifi";
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&gpio2 1 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&i2c {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <80000000>;
```

Comment:
Yes, this will work, because actually it will never reach 80MHz! See https://github.com/openwrt/openwrt/commit/a508ab1ac843c5a60a134bf3cdc8837aee6b8ccc#commitcomment-37769002
You can check the exact hardware frequency by changing this line from dev_dbg to dev_info.
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/patches-4.14/0042-SPI-ralink-add-Ralink-SoC-spi-driver.patch#L234
I guess 50MHz would suffice for this device.

Reply:
That makes sense, thanks for the info! My thought (by all means disagree!) ... store here the correct / real flash frequency. If it's limited by the CPU SPI, that's OK, that just means it's the actual frequency used (i.e. the flash allows up to 80 MHz, but of course you can choose to run it slower if needed).

Reply:
I don't think setting a high frequency just for the flash datasheet is a good practice. In ath79, there once were devices using 104MHz because the flash datasheet says it's okay, and run-test did not reveal any error. But after a new spi driver is introduced, the high frequency started to fail those devices. See ebc090e420d1.
I belive it's better to set a sensible frequency in the first place.

Reply:
I have no issue changing it - do you have a proposed value? I admit, if the flash datasheet calls out a value, I would have thought that would be the right thing to use, vs. a limit imposed by something other part on the board (as it gets a bit confusing later to explain the value ... LOL). But really no biggie here - what to change it to?

Thanks.

Reply:
You can try 50MHz. AFAIK it's the highest frequency possible on most mt7620 boards.

Reply:
OK, will do. I'll add a note to the commit also - that may at least help later if anyone is confused. New push coming.